parquet-hadoop/src/main/java/org/apache/parquet/format/converter/ParquetMetadataConverter.java

```diff
@@ -324,6 +346,92 @@ public static Statistics toParquetStatistics(
     return stats;
   }
 
+  public org.apache.parquet.column.statistics.Statistics fromParquetStatistics(
+      String createdBy, Statistics statistics, PrimitiveType type) {
+    SortOrder expectedOrder = isSignedOrderOkay(type) ? SortOrder.SIGNED : sortOrder(type);
+    return fromParquetStatisticsInternal(
+        createdBy, statistics, type.getPrimitiveTypeName(), expectedOrder);
+  }
+
+  enum SortOrder {
+    SIGNED,
+    UNSIGNED,
+    UNKNOWN
+  }
+
+  private boolean isSignedOrderOkay(PrimitiveType type) {
+    if (!useSignedStringMinMax) {
+      return false;
+    }
+
+    // even if the override is set, only return stats for string-ish types
+    if (type.getPrimitiveTypeName() != PrimitiveTypeName.BINARY) {
+      return false;
+    }
+    if (type.getOriginalType() == null) {
+      return true; // plain binary is okay
+    }
+    switch (type.getOriginalType()) {
+      case UTF8:
+      case ENUM:
+      case BSON:
+      case JSON:
+        return true;
+      default: // includes decimal
+        return false;
+    }
+  }
```

Comment:
I don't see why we restrict it further since the flag is already about lifting the restriction but that's fine with me if you feel we need this.

couple more comments:
- The name of the method could be clearer. What it actually does is controlling if we're going to override the result of sortOrder().

- The way the logic is organized here is a little hard to read.
If you define the accepted String types:
```
private static final Set<OriginalType> acceptedStringTypes = Collections.unmodifiableSet(new HashSet(Arrays.asList(
  UTF8, ENUM, JSON
)));
```
then it boils down to a boolean expression that I find easier to understand:
```
boolean overrideSortOrderToUnsigned(PrimitiveType type) {
  return useSignedStringMinMax 
    && (type.getPrimitiveTypeName() == BINARY) 
    && (type.getOriginalType() == null || acceptedStringTypes.contains(type.getOriginalType());
}
```